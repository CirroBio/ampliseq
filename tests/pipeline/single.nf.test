nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_single"
    tag "pipeline"

    test("Single-End") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/overall_summary.tsv")).match("overall_summary_tsv") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                path("$outputDir/barrnap/rrna.bac.gff"),
                                path("$outputDir/barrnap/rrna.euk.gff"),
                                path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/cutadapt/cutadapt_summary.tsv")).match("cutadapt") },
                { assert new File("$outputDir/cutadapt/1a_S103_L001_R1_001.trimmed.cutadapt.log").exists() },
                { assert new File("$outputDir/cutadapt/1_S103_L001_R1_001.trimmed.cutadapt.log").exists() },
                { assert new File("$outputDir/cutadapt/2a_S115_L001_R1_001.trimmed.cutadapt.log").exists() },
                { assert new File("$outputDir/cutadapt/2_S115_L001_R1_001.trimmed.cutadapt.log").exists() },
                { assert new File("$outputDir/cutadapt/assignTaxonomy.cutadapt.log").exists() },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                path("$outputDir/dada2/ASV_table.tsv"),
                                path("$outputDir/dada2/ref_taxonomy.rdp_18.txt"),
                                path("$outputDir/dada2/DADA2_stats.tsv"),
                                path("$outputDir/dada2/DADA2_table.rds"),
                                path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax.rdp_18.tsv").exists() },
                { assert new File("$outputDir/dada2/ASV_tax_species.rdp_18.tsv").exists() },
                { assert new File("$outputDir/fastqc/1a_S103_L001_R1_001_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/1_S103_L001_R1_001_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/2a_S115_L001_R1_001_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/2_S115_L001_R1_001_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/input/Samplesheet_single_end.tsv")).match("input") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt")).match("multiqc") }
            )
        }
    }
}
