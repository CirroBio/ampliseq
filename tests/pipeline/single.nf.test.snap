{
    "input": {
        "content": [
            "Samplesheet_single_end.tsv:md5,71bcb9920b1187571ba9e2a5759ee4a5"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    },
    "cutadapt": {
        "content": [
            "cutadapt_summary.tsv:md5,5e5bfa4a7324a44f6d9e3cb0978ca291"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    },
    "software_versions": {
        "content": [
            "{BARRNAP={barrnap=0.9}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, CUTADAPT_BASIC={cutadapt=3.4}, DADA2_DENOISING={R=4.1.1, dada2=1.22.0}, DADA2_FILTNTRIM={R=4.1.1, dada2=1.22.0}, DADA2_QUALITY1={R=4.1.1, ShortRead=1.52.0, dada2=1.22.0}, DADA2_TAXONOMY={R=4.1.1, dada2=1.22.0}, FASTQC={fastqc=0.11.9}, RENAME_RAW_DATA_FILES={sed=4.7}, Workflow={nf-core/ampliseq=2.6.0dev}}"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    },
    "overall_summary_tsv": {
        "content": [
            "overall_summary.tsv:md5,0feea9a92fde36cbf63dba6e63617c7e"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,d452ff8b8a306b52ffc6db7e4396c6db",
            "ASV_table.tsv:md5,a1226d8573fc0595161d4b2b5ac63cac",
            "ref_taxonomy.rdp_18.txt:md5,815c4fce9f3d1de019fb995a43fb66ed",
            "DADA2_stats.tsv:md5,8386cc209c1f64237deeec79f75b075b",
            "DADA2_table.rds:md5,aefd24f6ac2753a43baca19a93c4e2ee",
            "DADA2_table.tsv:md5,3e5280fd5b36c943c0148c4d5b50cb65"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,7ba69096ac8690260dcb2139ba2d9d82",
            "rrna.bac.gff:md5,15c5fa3b75684c42d60174b77fa87a09",
            "rrna.euk.gff:md5,1978664ae39a761f9082ae605c026182",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_fastqc.txt:md5,0ea2e6e2d327d66e778e9ff5d03d933b",
            "multiqc_general_stats.txt:md5,5040629a246bb3288879d3d30e9d6f40",
            "multiqc_cutadapt.txt:md5,48d079bea04fe93d260b980c15793a0c"
        ],
        "timestamp": "2023-05-28T20:35:33+0000"
    }
}